### PR TITLE
New package: mailspring-1.9.2

### PR DESCRIPTION
**Pull request by fanyx**
#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- Unable to build as of time of opening PR
- Needs assistance with fixing libcrypto.so.1.0.0 issue

### CODE REVIEW

**Comment by Johnnynator** [comment]
```
    SONAME: libcrypto.so.1.0.0 <-> UNKNOWN PKG PLEASE FIX!
```

Some binary does link against OpenSSL 1.0, which is not shipped by Void, is long End of Life and should not be used anymore.

You could try building these parts from source against OpenSSL 1.1, but that might be a lot of pain if I remember correctly.

**Comment by fanyx** [comment]
Yeah, i feel like the libcrypto 1.0.0 linking should not be in the software itself. Will open an issue with mailspring itself to see if that can be fixed.

For the time being, is there a way to disable the automatic shlibs generation and include libcrypto1.1 as a dep manually?

The app is running just fine if i install the files to my system by hand.

File: srcpkgs/mailspring/template [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Template file for 'mailspring'
+pkgname=mailspring
+version=1.9.2
+revision=1
+archs="x86_64"
+depends="glibc glib libgcc libstdc++ libX11 libxkbfile libsecret mit-krb5-libs libsasl libdb libXcomposite
+	libXcursor libXdamage libXext libXfixes libXi libXrender libXtst nss nspr dbus-libs gdk-pixbuf gtk+3 pango
+	atk cairo expat libXrandr libXScrnSaver alsa-lib at-spi2-atk at-spi2-core libcups libcrypto1.1"
```
Comment (ericonr) [inline srcpkgs/mailspring/template:8]: Most of these will be detected automatically by shlib scanning.

File: srcpkgs/mailspring/template [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Template file for 'mailspring'
+pkgname=mailspring
+version=1.9.2
+revision=1
+archs="x86_64"
```
Comment (ericonr) [inline srcpkgs/mailspring/template:5]: Building from source should at least enable us to remove this.

File: srcpkgs/mailspring/template [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	libXcursor libXdamage libXext libXfixes libXi libXrender libXtst nss nspr dbus-libs gdk-pixbuf gtk+3 pango
+	atk cairo expat libXrandr libXScrnSaver alsa-lib at-spi2-atk at-spi2-core libcups libcrypto1.1"
+short_desc="Standalone Mail client with advanced feature set"
+maintainer="fanyx <fanyx@pm.me>"
+license="custom:GPL-3.0-or-later"
+homepage="https://getmailspring.com/"
+distfiles="https://github.com/Foundry376/Mailspring/releases/download/${version}/mailspring-${version}-amd64.deb"
+checksum="a3bc365c77791673d8b7d8fc2081b2e8e80ab5d78d6ffa99f848971849f43544"
+noshlibprovides=yes
+nopie_files="/usr/lib/mailspring/resources/app.asar.unpacked/mailsync.bin"
```
Comment (ericonr) [inline srcpkgs/mailspring/template:16]: This should end up being PIE when built from source.

**Review by ericonr — COMMENTED** [review]
That an email application in 2021 is still using OpenSSL 1.0.0 at all is kind of concerning.

As is, I'm against inclusion of this in repos. It's open source, at least the C++ parts should be built by us.

**Comment by fanyx** [comment]
Agree.

I'll look into building it locally from source.
Closing this PR for now until i can successfully build it with OpenSSL 1.1.
Might just leave it out entirely, not a daily driver software anyways.

**Pull request closed without merging**